Source/WTF/wtf/StdLibExtras.h

```diff
@@ -781,8 +781,16 @@ template<typename OptionalType> auto valueOrDefault(OptionalType&& optionalValue
 template<typename T, typename U, std::size_t Extent>
 std::span<T, Extent> spanReinterpretCast(std::span<U, Extent> span)
 {
-    RELEASE_ASSERT(!(span.size_bytes() % sizeof(T)));
-    static_assert(std::is_const_v<T> || (!std::is_const_v<T> && !std::is_const_v<U>), "spanCast will not remove constness from source");
+    if constexpr (sizeof(U) % sizeof(T))
+        RELEASE_ASSERT(!(span.size_bytes() % sizeof(T)));
+
+    static_assert(std::is_const_v<T> || (!std::is_const_v<T> && !std::is_const_v<U>), "spanReinterpretCast will not remove constness from source");
+
+    if constexpr (std::is_same_v<T, std::byte>)
```

Comment:
I'm not sure these are worth the additional complexity for something that doesn't change anything.

Reply:
Let's instead add static_assert(!std::is_same<std::remove_const_t<T>, std::byte>, "Use std::as_bytes or std::as_writable_bytes instead")

Reply:
Via offline discussion, I decided to remove the `std::as_[writable_]bytes` early returns.